src/main/java/org/kohsuke/github/GHEventPayload.java

```diff
@@ -207,6 +208,164 @@ void wrapUp(GitHub root) {
         }
     }
 
+    /**
+     * Represents a repository object specifically for {@link Installation} and {@link InstallationRepositories}.
+     */
+    @SuppressFBWarnings(value = { "UWF_UNWRITTEN_FIELD" }, justification = "JSON API")
+    public static class InstallationRepository {
```

Comment:
Construct a new object to represent the repository since the payload only contains a small part of `GHRepository`; it would to safer to use. But not so sure where we should put it.

Reply:
The pattern has been to reuse existing classes.  Then there are two options:
1. **eagerly populate all the items via `wrapup()/lateBind()`** - this results in a multiple api requests up front but returns stable fully populated objects
2. **only refresh item when user requests data that is missing from partial** - can save a ton of requests but adds complexity to target class (missing fields need to be guarded with checks for partial state. 

In this case, the partial is so thin you should probably go with option 1.

Reply:
@XiongKezhi 
Hm, but is it possible to get an event for a Repository what the repository is private and can't be populated using the current credentials?

Reply:
> 
> 
> @XiongKezhi
> Hm, but is it possible to get an event for a Repository what the repository is private and can't be populated using the current credentials?

The repository installation events will be sent to the APP with [at least metadata read-only permission](https://developer.github.com/v3/apps/permissions/#metadata-permissions) to that repository, so shouldn't it be the users' responsibility if they use a client without the proper credentials instead of a client with the APP's JWT token?